doc/subsystems/networking/net-app-api.rst

```diff
@@ -0,0 +1,201 @@
+  connecting to other device.
+
+Next options are only available if IEEE 802.15.4 wireless network technology
+support is enabled.
+
+* **CONFIG_NET_APP_IEEE802154_DEV_NAME** tells the name of the IEEE 802.15.4
+  device.
+
+* **CONFIG_NET_APP_IEEE802154_PAN_ID** tells the used PAN identifier.
+  Note that the PAN id can be changed during runtime if needed.
+
+* **CONFIG_NET_APP_IEEE802154_CHANNEL** tells the used radio channel.
+  Note that the used channel can be changed during runtime if needed.
+
+* **CONFIG_NET_APP_IEEE802154_RADIO_TX_POWER** tells the initial radio
+  TX power level. The used TX power level can be changed during runtime
+  if needed.
+
+* **CONFIG_NET_APP_IEEE802154_SECURITY_KEY** tells the initially used
+  security key. The used security key can be changed during runtime
+  if needed.
+
+* **CONFIG_NET_APP_IEEE802154_SECURITY_KEY_MODE** tells the initially used
+  security key mode. The used security key mode can be changed during runtime
+  if needed.
+
+* **CONFIG_NET_APP_IEEE802154_SECURITY_LEVEL** tells the initially used
+  security level. The used security level can be changed during runtime
+  if needed.
+
+Client / Server Applications
+############################
+
+The net-app API provides functions that enable the application to create
+a client / server type applications easily. If needed, the applications can
+have the communication secured by TLS (for TCP connections) or DTLS (for
+UDP connections) automatically.
+
+A simple **TCP server** application would consists of following net-app API
+function calls:
+
+* `net_app_init_tcp_server()` will configure used TCP port and local address.
+
+* `net_app_set_cb()` will configure callback functions that are called
+  when for example new data is received.
+
+* `net_app_server_tls()` will optionally setup the system for secured
+  connections. To enable the TLS server, one needs to also call the
+  `net_app_server_tls_enable()` function.
+
+* `net_app_listen()` will start to listen new client connections.
+
+Creating a **UDP server** is also very easy:
+
+* `net_app_init_udp_server()` will configure used UDP port and local address.
+
+* `net_app_set_cb()` will configure callback functions that are called
+  when for example new data is received.
+
+* `net_app_server_tls()` will optionally setup the system for secured
+  connections. To enable the TLS server, one needs to also call the
+  `net_app_server_tls_enable()` function.
+
+* `net_app_listen()` will start to listen new client connections.
+
+If the server wants to stop listening connections, it can call
+`net_app_release()`. After this application will need to call the server
+init function again to start listening incoming connections.
+
+For TLS/DTLS connections, the server can be disabled by a call to
+`net_app_server_tls_disable()` function. There are separate enable/disable
+functions for TLS support as we need to have a separate crypto thread that
```

Comment:
Better would be, "... for TLS support because we need a separate crypto thread for calling mbedtls crypto API functions."   (I think)